File: src/main/java/no/nav/sosialhjelp/soknad/web/rest/ressurser/dialog/SistInnsendteSoknadRessurs.java
package no.nav.sosialhjelp.soknad.web.rest.ressurser.dialog;

import no.nav.security.token.support.core.api.ProtectedWithClaims;
import no.nav.sosialhjelp.metrics.aspects.Timed;
import no.nav.sosialhjelp.soknad.business.service.dialog.SistInnsendteSoknadService;
import no.nav.sosialhjelp.soknad.domain.model.oidc.SubjectHandler;
import no.nav.sosialhjelp.soknad.web.rest.ressurser.dialog.dto.SistInnsendteSoknadDto;
import org.springframework.stereotype.Controller;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;

import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
import static no.nav.sosialhjelp.soknad.web.utils.Constants.CLAIM_ACR_LEVEL_4;
import static no.nav.sosialhjelp.soknad.web.utils.Constants.TOKENX;

@Controller
@ProtectedWithClaims(issuer = TOKENX, claimMap = {CLAIM_ACR_LEVEL_4})
@Path("/dialog")
@Produces(APPLICATION_JSON)
@Timed
public class SistInnsendteSoknadRessurs {

    private final SistInnsendteSoknadService nyligInnsendteSoknaderService;

    public SistInnsendteSoknadRessurs(
            SistInnsendteSoknadService nyligInnsendteSoknaderService
    ) {
        this.nyligInnsendteSoknaderService = nyligInnsendteSoknaderService;
    }

    @GET
    @Path("/sistInnsendteSoknad")
    public SistInnsendteSoknadDto hentNyligInnsendteSoknader() {
        var fnr = SubjectHandler.getUserId();
        return nyligInnsendteSoknaderService.hentSistInnsendteSoknad(fnr).orElse(null);
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/api/dialog/SistInnsendteSoknadRessurs.kt
package no.nav.sosialhjelp.soknad.api.dialog

import no.nav.security.token.support.core.api.ProtectedWithClaims
import no.nav.sosialhjelp.soknad.api.dialog.dto.SistInnsendteSoknadDto
import no.nav.sosialhjelp.soknad.app.Constants.CLAIM_ACR_LEVEL_4
import no.nav.sosialhjelp.soknad.app.Constants.TOKENX
import no.nav.sosialhjelp.soknad.app.subjecthandler.SubjectHandlerUtils
import org.springframework.stereotype.Controller
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Controller
@ProtectedWithClaims(issuer = TOKENX, claimMap = [CLAIM_ACR_LEVEL_4])
@Path("/dialog")
@Produces(MediaType.APPLICATION_JSON)
open class SistInnsendteSoknadRessurs(
    private val sistInnsendteSoknadService: SistInnsendteSoknadService
) {
    @GET
    @Path("/sistInnsendteSoknad")
    open fun hentSistInnsendteSoknad(): SistInnsendteSoknadDto? {
        val fnr = SubjectHandlerUtils.getUserIdFromToken()
        return sistInnsendteSoknadService.hentSistInnsendteSoknad(fnr)
    }
}
